src/compiler/crystal/config.cr

```diff
@@ -30,27 +30,19 @@ module Crystal
     end
 
     private def self.compute_version_and_sha
-      # Set explicitly: 0.0.0, ci, HEAD, whatever
-      config_version = {{env("CRYSTAL_CONFIG_VERSION")}}
-      return {config_version, nil} if config_version
-
-      git_version = {{`(git describe --tags --long --always 2>/dev/null) || true`.stringify.chomp}}
-
-      # Failed git and no explicit version set: ""
-      # We inherit the version of the compiler building us for now.
-      return { {{Crystal::VERSION}}, nil } if git_version.empty?
-
-      # Shallow clone with no tag in reach: abcd123
-      # We assume being compiled with the latest released compiler
-      return {"#{{{Crystal::VERSION}}}+?", git_version} unless git_version.includes? '-'
-
-      # On release: 0.0.0-0-gabcd123
-      # Ahead of last release: 0.0.0-42-gabcd123
-      tag, commits, sha = git_version.split('-')
-      sha = sha[1..-1]                                # Strip g
-      tag = "#{tag}+#{commits}" unless commits == "0" # Reappend commits since release unless we hit it exactly
-
-      {tag, sha}
+      config_version = {% if flag?(:windows) %}
+                       {{ `type #{__DIR__}/../../../VERSION`.stringify.chomp }}
+                       {% else %}
+                       {{ `cat #{__DIR__}/../../../VERSION`.stringify.chomp }}
+                       {% end %}
+      git_describe = {{ `(git describe --tags --long --always 2>/dev/null) || true`.stringify.chomp }}
```

Comment:
`git rev-parse --short HEAD 2> /dev/null` returns the sha directly. No need to parse the describe string.